### PR TITLE
feat(validation): row-level definitional checks for CFB + NFL pbp

### PR DESCRIPTION
**Pull request by saiemgilani**
## Summary

Adds a `definitional` check module to the Tier-1 validation harness: per-dataset rule tables of named polars boolean violation expressions, evaluated in a single select pass, emitting one `Finding` per fired rule with the violation count and up to 3 sample rows for triage. Wired into the CLI's `_CHECKS` dispatch.

**Rule families** (22 CFB + 57 NFL rules):
- **Flag algebra** — rush/pass mutual exclusivity, complete/incomplete exclusivity, TD subtype algebra (≤1 subtype, subtype⇒touchdown⇒sp), FG outcome algebra, fumble_lost⇒fumble, kneel/spike + kickoff/punt exclusivity, the producer's exact `passing_down` definition.
- **Derived-value identities** — epa/wpa snapshot diffs, score_differential, clock hierarchy (scoped to regulation), down/distance/yardline/timeout/clock bounds, strictly-increasing `game_play_number` per game.
- **Participant name/id relationships** — name/id pairs never half-populate, flag-iff-participant (passer/rusher/td/penalty/timeout), receiver required on completions but deliberately allowed null on incompletes (throwaways), extracted names must appear verbatim in the play text.
- **Content-level type rules** — GSIS id format, nflverse game_id format, numeric-string drive.id (structurally subsumes the `"123.0"` float-artifact and empty-string id bug classes), clean-string (no empty/padded values), posteam/defteam participant + distinctness.

**Severity calibration is measured, not guessed**: every rule was run against the published parquet; rules ship as ERROR only where the invariant held 100% (e.g. all 108,828 NFL ids GSIS-format, all names verbatim in desc). Measured exceptions are WARN `needs_judgment` routed to Tier-2 triage, each with the evidence in a code comment (CFB epa≠ep_after−ep_before on ~1.2% of rows while the wpa twin holds exactly; fumble-return-TD passers; nickname-vs-formal name divergence at 97.7%).

Null semantics are load-bearing: violation expressions null-propagate so identity rules automatically apply only to fully-populated rows; a `SchemaError` fallback handles degenerate Null-dtype series (all-null single-file reads) without masking schema_contract's own dtype findings.

## Testing
- 17 new offline tests in `tests/contracts/test_definitional.py` (117 total contracts suite green)
- Calibration runs on both published datasets: zero ERRORs, 5 designed triage WARNs

## Summary by Sourcery

Introduce row-level definitional validation checks for CFB and NFL play-by-play datasets and integrate them into the Tier-1 validation CLI.

New Features:
- Add a definitional check module that evaluates per-dataset boolean rule tables over play-by-play data and emits findings with counts and sample rows.
- Define comprehensive CFB and NFL rule sets covering flag algebra, derived-value identities, participant relationships, and content-level type constraints.
- Integrate the definitional checks into the validation CLI check dispatch sequence.

Tests:
- Add a dedicated test suite for definitional checks covering rule selection, null semantics, sample attachment, severity calibration, and dataset-specific invariants.

## Summary by CodeRabbit

- **New Features**
  - Added definitional validation for college and professional football play-by-play datasets.
  - Checks data consistency, statistical relationships, participant fields, identifiers, string formatting, and game-state rules.
  - Validation results include violation counts, sample records, severity levels, and review guidance.
  - Automatically skips checks when required fields are unavailable.

- **Tests**
  - Added comprehensive coverage for validation rules, edge cases, exceptions, and dataset-specific behavior.

### CODE REVIEW

File: tools/validation/checks/definitional.py [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+            )
+        ),
+        "passing_down must equal (down 2 & dist>=8) | (down 3/4 & dist>=5) on scrimmage plays",
+    ),
+    Rule(
+        "game_play_number_strictly_increasing",
+        ("game_id", "game_play_number"),
+        _c("game_play_number").diff().over("game_id") <= 0,
+        "game_play_number must strictly increase within a game",
+    ),
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tools/validation/checks/definitional.py:222]: Partially applied in 72e782f5: the ordering precondition is now documented on the rule (comment + description). Declined the sort_by option — play-order-within-game is the harness dataset contract (same precondition boundary_leakage documents), and this rule exists to VERIFY the stored order; sorting by game_play_number first would make it tautological.

File: tools/validation/checks/definitional.py [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+        # Null) breaks schema resolution for string-op rules. Fall back to
+        # per-rule evaluation and skip the unresolvable ones — the dtype
+        # divergence itself is schema_contract's finding, not a definitional
+        # violation.
+        counts = {}
+        for r in rules:
+            try:
+                counts[r.name] = frame.select(r.violation.sum().cast(pl.Int64)).item()
+            except pl.exceptions.SchemaError:
+                counts[r.name] = 0
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tools/validation/checks/definitional.py:609]: Fixed in 72e782f5: both except clauses now catch pl.exceptions.PolarsError, so one degenerate column cannot suppress the other rules' findings. Rule-authoring errors remain covered by the per-rule unit tests rather than this handler.
